File: Java - Microservices with Spring Boot, Spring Cloud/Glide-Smart-Bike-Rental-System-main/glide-frontend/src/services/rideService.js
import api from "./api";

export const startRide = async (reservationId) => {

    const response = await api.post(
        `/api/rides/start/${reservationId}`
    );

    return response.data;
};

export const getMyRides = async () => {

    const response = await api.get("/api/rides/my");

    return response.data;
};

export const endRide = async (rideId, stationId) => {

    const response = await api.post(
        `/api/rides/end/${rideId}?stationId=${stationId}`
    );

    return response.data;
};

File: Java - Microservices with Spring Boot, Spring Cloud/Glide-Smart-Bike-Rental-System-main/glide-backend/src/main/java/com/glide/service/RideService.java
package com.glide.service;

import com.glide.dto.RideResponse;

import java.util.List;

public interface RideService {

    RideResponse startRide(Long reservationId);

    RideResponse endRide(Long rideId, Long stationId);

    List<RideResponse> getMyRides();

}